control-plane/pkg/reconciler/sink/controller.go

```diff
@@ -89,5 +92,12 @@ func NewController(ctx context.Context, _ configmap.Watcher, configs *config.Env
 		},
 	})
 
+	reconciler.SecretTracker = tracker.New(impl.EnqueueKey, controller.GetTrackerLease(ctx))
+	secretinformer.Get(ctx).Informer().AddEventHandler(controller.HandleAll(reconciler.SecretTracker.OnChanged))
+
+	sinkInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		DeleteFunc: reconciler.OnDeleteObserver,
```

Comment:
Why that? Can't you just invoke `OnDeleteObserver` in the FinalizeKind of the broker reconciler?

Reply:
I started with your proposed approach and then migrated to this one since this is the way used in other places in eventing.
ATM, they should be equivalent. 
However, given this issue: https://github.com/knative/pkg/issues/1956, I think it's safer to call it regardless of being the leader or not.

Reply:
Can you add a TODO, in order to remember in future that we need to port this code to `FinalizeKind` after pkg#1956 is solved? The FinalizeKind approach is still more idiomatic

Reply:
For example:
time 1: p is the leader for x -> reconcile x
time 2: p is not the leader for x anymore
time 3: delete x
time 4: no FinilizeKind called on x in p (this is a leak)

Calling it outside FinilizeKind makes sure that the reconciler doesn't leak.

Some usages: https://github.com/knative/serving/blob/b7a7c18bb5f6c14ae73c9dda1a817ded1ae879b0/pkg/reconciler/route/controller.go#L108-L110

Reply:
> ATM, they should be equivalent.

This is not true.